erpnext/hr/doctype/leave_type/leave_type.py

```diff
@@ -2,9 +2,23 @@
 # License: GNU General Public License v3. See license.txt
 
 from __future__ import unicode_literals
+import calendar
 import frappe
+from datetime import datetime
+from frappe.utils import today
+from frappe import _
 
 from frappe.model.document import Document
 
 class LeaveType(Document):
-	pass
\ No newline at end of file
+	def validate(self):
+		if self.is_lwp:
+			leave_allocation = frappe.get_all("Leave Allocation", filters={
+				'leave_type': self.name,
+				'from_date': ("<=", today()),
+				'to_date': (">=", today())
+			}, fields=['name'])
+			leave_allocation = [l['name'] for l in leave_allocation]
+			frappe.db("""select name from `tabLeave Allocation` where leave_type=%s""", (self.name))
```

Comment:
`frappe.db("""select name from `tabLeave Allocation` where leave_type=%s""", (self.name))
`

Shouldn't it be `frappe.db.sql`, or should it be removed?

Reply:
added changes